src/rest/subhandlers/rest_meta_info.erl

```diff
@@ -46,43 +46,85 @@ init(Req0, State) ->
 allowed_methods(Req, State) ->
     {[<<"GET">>, <<"POST">>, <<"PUT">>], Req, State}.
 
-content_types_provided(Req, State) ->
-    {[{<<"application/json">>, to_json}], Req, State}.
-
 content_types_accepted(Req, State) ->
     {[{<<"application/json">>, from_json}], Req, State}.
 
-%% @todo implement case for dependent_zone
+content_types_provided(Req, State) ->
+    {[{<<"application/json">>, to_json}], Req, State}.
+
+%% @todo implement case for dependent_zones, if needed
 resource_exists(Req, State) ->
+    {true, Req, State}.
+
+%% POST/PUT handler
+from_json(Req, State) ->
     Result = case maps:get(op, State) of
-                 dependent_zone -> some_logic;
-                 _ -> true
+                 instrumentation ->
+                     {ok, Bool} = parse_boolean(maps:get(enabled, State)),
+                     set_instrumentation(Bool);
+                 indexation ->
+                     {ok, Bool} = parse_boolean(maps:get(enabled, State)),
+                     set_indexation(Bool);
+                 dependent_zones ->
+                     case parse_dependent_zones(maps:get(body, State)) of
+                         {ok, List} -> set_dependent_zones(List);
+                         {error, _} -> false
+                     end
              end,
     {Result, Req, State}.
 
+%% GET handler
+to_json(Req, State) ->
+    %% Result should be boolean
+    Result = case maps:get(op, State) of
+                 health_check -> true;
+                 index_ready -> get_index_ready();
+                 instrumentation -> get_instrumentation()
+             end,
+    {json_utils:encode(Result), Req, State}.
+
 
 %%%---------------------------
 %% internal functions
 %%%---------------------------
 
-%% POST/PUT handler
+%% @todo
+set_instrumentation(_Bool) ->
+    true.
 
 %% @todo
-from_json(Req, State) ->
-    Result = case maps:get(op, State) of
-                 instrumentation -> true;
-                 indexation -> true;
-                 dependent_zone -> {true, dependentZonesDto}
-             end,
-    {Result, Req, State}.
+get_instrumentation() ->
+    false.
 
-%% GET handler
+%% @todo
+set_indexation(_Bool) ->
+    true.
 
 %% @todo
-to_json(Req, State) ->
-    Result = case maps:get(op, State) of
-                 health_check -> true;
-                 index_ready -> some_boolean;
-                 instrumentation -> some_boolean
-             end,
-    {Result, Req, State}.
+get_index_ready() ->
```

Comment:
Mało deskryptywna nazwa funkcji. Zmień to, bo nie mam pojęcia, co to ma robić

Reply:
Też nie mam pojęcia co to ma robić XDDD.
Po prostu Kamil ma to u siebie, na razie zaimplementowałem tylko rest api do tego, a potem się dokształce jak to działa

Reply:
Dobra, już wiem.

Tak więc, zmieniłem nazwę i dodałem komentarz do tego